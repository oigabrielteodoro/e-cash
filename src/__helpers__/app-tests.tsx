import React, { useEffect, ReactElement, ReactNode } from 'react'
import { render as rtlRender } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import {
  DefaultOptions,
  QueryCache,
  QueryClient,
  QueryClientProvider,
} from 'react-query'
import { MemoryRouter, Routes, Route } from 'react-router-dom'
import { generateId } from 'lib'
import nock from 'nock'

import { pipe } from 'fp-ts/function'
import { TaskEither, map, mapLeft } from 'fp-ts/TaskEither'

import { AppLayout, NotificationContainer } from 'ui'
import { clearToken, queryConfigDefault, setToken } from 'client'

type Options = {
  initialRoute?: string
  initialRoutePath?: string
  authenticated?: boolean
  queryClient?: QueryClient
  routePaths?: string[]
}

type Props = {
  children?: ReactNode
  client?: QueryClient
}

type Callback = (result: unknown) => unknown
type MapAll = (
  fn: Callback,
) => (data: TaskEither<unknown, unknown>) => TaskEither<unknown, unknown>

const queryCache = new QueryCache()
const defaultQueryClient = new QueryClient({
  defaultOptions: queryConfigDefault as DefaultOptions,
  queryCache: queryCache,
})

export function ReactQueryWrapper({
  children,
  client = defaultQueryClient,
}: Props) {
  return <QueryClientProvider client={client}>{children}</QueryClientProvider>
}

export function ReactRouterWrapper({
  initialRoute = '/',
  children,
  routePaths,
  authenticated = true,
}: Options & Pick<Props, 'children'>) {
  function MockedPage({ title }: { title: string }) {
    if (authenticated) {
      return (
        <AppLayout>
          <AppLayout.Content>Page {title}</AppLayout.Content>
        </AppLayout>
      )
    }

    return <h1>Page {title}</h1>
  }

  return (
    <MemoryRouter initialEntries={[initialRoute]}>
      <Routes>
        <Route path={initialRoute} element={children} />
        {routePaths?.map((routePath) => (
          <Route
            key={routePath}
            path={routePath}
            element={<MockedPage title={routePath} />}
          />
        ))}
      </Routes>
    </MemoryRouter>
  )
}

export function ReactHookWrapper({
  initialRoute = '/',
  children,
  routePaths,
  authenticated = true,
  client = defaultQueryClient,
}: Options & Props) {
  return (
    <ReactQueryWrapper client={client}>
      <ReactRouterWrapper
        initialRoute={initialRoute}
        routePaths={routePaths}
        authenticated={authenticated}
      >
        {children}
      </ReactRouterWrapper>
    </ReactQueryWrapper>
  )
}

function render(
  ui: ReactElement,
  {
    initialRoute = '/',
    queryClient = defaultQueryClient,
    authenticated = true,
    routePaths,
  }: Options = {},
) {
  function Wrapper({ children }: Props) {
    useEffect(() => {
      if (authenticated) {
        setToken(generateId())

        return () => {
          clearToken()
        }
      }
    }, [])

    return (
      <ReactQueryWrapper client={queryClient}>
        <ReactRouterWrapper
          initialRoute={initialRoute}
          routePaths={routePaths}
          authenticated={authenticated}
        >
          {children}
        </ReactRouterWrapper>

        <NotificationContainer />
      </ReactQueryWrapper>
    )
  }

  return {
    ...rtlRender(ui, {
      wrapper: Wrapper,
    }),
  }
}

export function unsafe<T>(value: unknown) {
  return value as T
}

export const mapAll: MapAll = (fn) => (data) => {
  return pipe(data, map(fn), mapLeft(fn))
}

export function getErrorMessage(errors: unknown): string {
  return Array.isArray(errors) ? errors[0].message : ''
}

export * from '@testing-library/react'
export {
  render,
  userEvent,
  nock,
  defaultQueryClient as queryClient,
  queryCache,
}
