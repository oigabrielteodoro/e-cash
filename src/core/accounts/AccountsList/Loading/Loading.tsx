import React from 'react'
import { Col, Row } from 'ui'

import { AccountCard } from 'core/accounts/AccountCard'

export function Loading() {
  return (
    <Row gutter={[24, 24]}>
      <Col md={24} lg={12} xlg={8}>
        <AccountCard.Skeleton />
      </Col>
      <Col md={24} lg={12} xlg={8}>
        <AccountCard.Skeleton />
      </Col>
      <Col md={24} lg={12} xlg={8}>
        <AccountCard.Skeleton />
      </Col>
      <Col md={24} lg={12} xlg={8}>
        <AccountCard.Skeleton />
      </Col>
    </Row>
  )
}
