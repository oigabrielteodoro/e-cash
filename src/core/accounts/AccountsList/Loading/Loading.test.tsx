import React from 'react'
import { render } from '__helpers__/app-tests'

import { Loading } from './Loading'

describe('Loading', () => {
  it('should be able match snapshot', () => {
    const { container } = render(<Loading />)

    expect(container).toMatchSnapshot()
  })
})
