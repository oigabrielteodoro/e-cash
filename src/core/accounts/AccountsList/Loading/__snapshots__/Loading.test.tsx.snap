// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should be able match snapshot 1`] = `
.c8 {
  -webkit-animation: ePEqAU 1.2s ease-in-out infinite;
  animation: ePEqAU 1.2s ease-in-out infinite;
  background-image: linear-gradient( 90deg,rgba(0,0,0,0),#fff,rgba(0,0,0,0) );
  background-color: #b3b3b3;
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  min-width: 3.2rem;
  height: 1.6rem;
  opacity: 0.1;
  width: 50%;
  width: 5.2rem;
  height: 5.2rem;
  border-radius: 50%;
}

.c8 + span {
  margin-top: 8px;
}

.c10 {
  -webkit-animation: ePEqAU 1.2s ease-in-out infinite;
  animation: ePEqAU 1.2s ease-in-out infinite;
  background-image: linear-gradient( 90deg,rgba(0,0,0,0),#fff,rgba(0,0,0,0) );
  background-color: #b3b3b3;
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  min-width: 3.2rem;
  height: 1.6rem;
  opacity: 0.1;
  width: 50%;
  border-radius: 0.5rem;
}

.c10 + span {
  margin-top: 8px;
}

.c12 {
  -webkit-animation: ePEqAU 1.2s ease-in-out infinite;
  animation: ePEqAU 1.2s ease-in-out infinite;
  background-image: linear-gradient( 90deg,rgba(0,0,0,0),#fff,rgba(0,0,0,0) );
  background-color: #b3b3b3;
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  min-width: 3.2rem;
  height: 1.6rem;
  opacity: 0.1;
  width: 25%;
  border-radius: 0.5rem;
}

.c12 + span {
  margin-top: 8px;
}

.c6 {
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  row-gap: 24px;
  margin: 0 -12px;
}

.c0 .c1 {
  padding: 0 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c3 {
  padding: 2.4rem;
  background: #ffffff;
  border-radius: 0.5rem;
  border: 0.1rem solid #cfd7e3;
}

.c4 {
  width: 100%;
  padding: 2.4rem 1.2rem;
  -webkit-transition: 300ms;
  transition: 300ms;
  cursor: pointer;
}

.c4:hover {
  -webkit-transform: translateY(-1rem);
  -ms-transform: translateY(-1rem);
  transform: translateY(-1rem);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 1.6rem;
  text-align: left;
}

.c9 strong {
  width: 25rem;
  font-weight: 600;
  font-size: 1.6rem;
  text-overflow: ellipsis;
}

.c9 small {
  width: 25rem;
  text-overflow: ellipsis;
  font-size: 1.4rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 1.6rem;
  width: 100%;
}

.c11 span {
  color: #a1a7ae;
  font-size: 1.2rem;
}

@media screen and (max-width:768px) {
  .c2 {
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
    max-width: 100%;
  }
}

@media screen and (min-width:768px) and (max-width:1441px) {
  .c2 {
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
    max-width: 50%;
  }
}

@media screen and (min-width:1441px) {
  .c2 {
    -webkit-flex: 0 0 33.33%;
    -ms-flex: 0 0 33.33%;
    flex: 0 0 33.33%;
    max-width: 33.33%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <div
            class="c1 c6"
            span="24"
          >
            <div
              class="c5"
            >
              <div
                class="c7"
              >
                <span
                  class="c8"
                  data-testid="ShimmerEffect"
                  style="border-radius: 0.5rem; height: 4.2rem; min-width: 4.2rem;"
                />
                <div
                  class="c5"
                  style="justify-content: space-between; width: 100%;"
                  width="100%"
                >
                  <div
                    class="c9"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 20rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                  <div
                    class="c5"
                    style="flex-direction: column;"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="c11"
              >
                <span
                  class="c12"
                  data-testid="ShimmerEffect"
                />
                <span
                  class="c10"
                  data-testid="ShimmerEffect"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <div
            class="c1 c6"
            span="24"
          >
            <div
              class="c5"
            >
              <div
                class="c7"
              >
                <span
                  class="c8"
                  data-testid="ShimmerEffect"
                  style="border-radius: 0.5rem; height: 4.2rem; min-width: 4.2rem;"
                />
                <div
                  class="c5"
                  style="justify-content: space-between; width: 100%;"
                  width="100%"
                >
                  <div
                    class="c9"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 20rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                  <div
                    class="c5"
                    style="flex-direction: column;"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="c11"
              >
                <span
                  class="c12"
                  data-testid="ShimmerEffect"
                />
                <span
                  class="c10"
                  data-testid="ShimmerEffect"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <div
            class="c1 c6"
            span="24"
          >
            <div
              class="c5"
            >
              <div
                class="c7"
              >
                <span
                  class="c8"
                  data-testid="ShimmerEffect"
                  style="border-radius: 0.5rem; height: 4.2rem; min-width: 4.2rem;"
                />
                <div
                  class="c5"
                  style="justify-content: space-between; width: 100%;"
                  width="100%"
                >
                  <div
                    class="c9"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 20rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                  <div
                    class="c5"
                    style="flex-direction: column;"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="c11"
              >
                <span
                  class="c12"
                  data-testid="ShimmerEffect"
                />
                <span
                  class="c10"
                  data-testid="ShimmerEffect"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c1 c2"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c5"
        >
          <div
            class="c1 c6"
            span="24"
          >
            <div
              class="c5"
            >
              <div
                class="c7"
              >
                <span
                  class="c8"
                  data-testid="ShimmerEffect"
                  style="border-radius: 0.5rem; height: 4.2rem; min-width: 4.2rem;"
                />
                <div
                  class="c5"
                  style="justify-content: space-between; width: 100%;"
                  width="100%"
                >
                  <div
                    class="c9"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 20rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                  <div
                    class="c5"
                    style="flex-direction: column;"
                  >
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                    <span
                      class="c10"
                      data-testid="ShimmerEffect"
                      style="width: 10rem;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="c11"
              >
                <span
                  class="c12"
                  data-testid="ShimmerEffect"
                />
                <span
                  class="c10"
                  data-testid="ShimmerEffect"
                  style="margin: 0px;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
